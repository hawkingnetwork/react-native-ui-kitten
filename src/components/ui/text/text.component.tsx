/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

import React from 'react';
import {
  Text as RNText,
  TextProps as RNTextProps,
} from 'react-native';
import {
  styled,
  StyledComponentProps,
} from '@kitten/theme';

type ChildElement = string | TextElement;

export interface TextProps extends StyledComponentProps, RNTextProps {
  category?: string;
  status?: string;
  children?: ChildElement | ChildElement[];
}

export type TextElement = React.ReactElement<TextProps>;

/**
 * Styled `Text` component.
 *
 * @extends React.Component
 *
 * @property {string} appearance - Determines the appearance of the component.
 * Can be `default`, `alternative`, `hint`.
 * Default is `default`.
 *
 * @property {string} status - Determines the status of the component.
 * Can be `basic`, `primary`, `success`, `info`, `warning`, `danger` or `control`.
 *
 * @property {string} category - Determines the category of the component.
 * Can be `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `s1`, `s2`, `p1`, `p2`, `c1`, `c2`, `label`.
 * Default is `p1`.
 *
 * @property {string | ReactElement<TextProps>} children - Determines text of the component.
 *
 * @property {TextProps} ...TextProps - Any props applied to Text component.
 *
 * @overview-example TextSimpleUsage
 *
 * @overview-example TextAppearances
 *
 * @overview-example TextCategories
 *
 * @overview-example TextStatuses
 *
 * @example TextInlineStyling
 */
export class TextComponent extends React.Component<TextProps> {

  static styledComponentName: string = 'Text';

  public render(): React.ReactElement<RNTextProps> {
    const { themedStyle, style, ...derivedProps } = this.props;

    return (
      <RNText
        {...derivedProps}
        style={[themedStyle, style]}
      />
    );
  }
}

export const Text = styled<TextProps>(TextComponent);
