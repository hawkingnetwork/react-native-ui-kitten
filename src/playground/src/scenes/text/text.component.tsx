import React from 'react';
import { ShowcaseContainer } from '@pg/components/showcaseContainer.component';
import { TextShowcase } from './textShowcase.component';
import {
  textSettings,
  textShowcase,
} from './type';

export const TextScreen = ({ navigation }): React.ReactElement => (
  <ShowcaseContainer
    showcase={textShowcase}
    settings={textSettings}
    renderItem={TextShowcase}
    onBackPress={() => navigation.goBack()}
  />
);
