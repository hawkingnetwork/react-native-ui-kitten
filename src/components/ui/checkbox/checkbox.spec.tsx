import React from 'react';
import { TouchableOpacity } from 'react-native';
import {
  render,
  fireEvent,
  RenderAPI,
  shallow,
  waitForElement,
} from 'react-native-testing-library';
import { ReactTestInstance } from 'react-test-renderer';
import {
  ApplicationProvider,
  ApplicationProviderProps,
} from '@kitten/theme';
import {
  CheckBox,
  CheckBoxProps,
} from './checkbox.component';
import { Text } from '../text/text.component';
import {
  mapping,
  theme,
} from '../support/tests';

const Mock = (props?: CheckBoxProps): React.ReactElement<ApplicationProviderProps> => {
  return (
    <ApplicationProvider
      mapping={mapping}
      theme={theme}>
      <CheckBox {...props} />
    </ApplicationProvider>
  );
};

const renderComponent = (props?: CheckBoxProps): RenderAPI => {
  return render(
    <Mock {...props}/>,
  );
};

describe('@checkbox matches snapshots', () => {

  it('* default', () => {
    const component: RenderAPI = renderComponent();
    const { output } = shallow(component.getByType(CheckBox));

    expect(output).toMatchSnapshot();
  });

  it('* checked.disabled', () => {
    const component: RenderAPI = renderComponent({
      checked: true,
      disabled: true,
    });
    const { output } = shallow(component.getByType(CheckBox));

    expect(output).toMatchSnapshot();
  });

  it('* active', async () => {
    const component: RenderAPI = renderComponent();

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(CheckBox);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(CheckBox);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('default');
  });

  it('* with text', () => {
    const text: string = 'Text';
    const component: RenderAPI = renderComponent({
      checked: true,
      text: text,
      textStyle: {
        fontSize: 18,
        color: 'red',
      },
    });
    const { output } = shallow(component.getByType(CheckBox));
    expect(component.getByType(Text).props.children).toBe(text);
    expect(output).toMatchSnapshot();
  });

});

describe('@checkbox: component checks', () => {

  it('* emits onChange with correct args', () => {
    const onChange = jest.fn();

    const component: RenderAPI = renderComponent({
      checked: true,
      onChange: onChange,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(onChange).toBeCalledWith(false, false);
  });

  it('* touchable other props', () => {
    const onPressIn = jest.fn();
    const onPressOut = jest.fn();
    const component: RenderAPI = renderComponent({
      onPressIn: onPressIn,
      onPressOut: onPressOut,
    });

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');
    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    expect(onPressIn).toHaveBeenCalled();
    expect(onPressOut).toHaveBeenCalled();
  });

});
