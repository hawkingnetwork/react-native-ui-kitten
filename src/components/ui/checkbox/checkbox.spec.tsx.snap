// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@checkbox matches snapshots * active 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@checkbox matches snapshots * active: default 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@checkbox matches snapshots * checked.disabled 1`] = `
<Wrapper
  checked={true}
  disabled={true}
  forwardedRef={null}
/>
`;

exports[`@checkbox matches snapshots * default 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@checkbox matches snapshots * with text 1`] = `
<Wrapper
  checked={true}
  forwardedRef={null}
  text="Text"
  textStyle={
    Object {
      "color": "red",
      "fontSize": 18,
    }
  }
/>
`;
